pkg/apis/minio.min.io/v1/types.go

```diff
@@ -244,6 +246,14 @@ type AuditConfig struct {
 	DiskCapacityGB *int `json:"diskCapacityGB,omitempty"`
 }
 
+// PrometheusConfig defines configuration for Prometheus metrics server
+type PrometheusConfig struct {
+	// DiskCapacityGB defines the disk capacity in GB available to the
+	// Prometheus server
+	// +optional
+	DiskCapacityDB *int `json:"diskCapacityGB,omitempty"`
```

Comment:
Can we add support for `NodeSelector`, `Annotations`, `Labels` and `Resources`  for the Prometheus pods?

To configure the storage we should expose `corev1.PersistentVolumeClaim` so that users can configure storage class and any other CSI related configurations